src/Redux/Reducers/authSlice.js

```diff
@@ -0,0 +1,57 @@
+import {createAsyncThunk, createSlice} from "@reduxjs/toolkit";
+import axios from "axios";
+import { AlertToast,SuccessToast } from "../../components/toasts";
+
+const initialState = {
+    currentUser:{},
+};
+
+export const login = createAsyncThunk("/auth/login", async (e,userDetail) =>{
+    try {
+        const response = await axios.post(`api/auth/login`, e.target.value === "credentialLogin" ? {
+            username: "adarshbalika",
+            password: "adarshBalika123",
+        } : JSON.stringify(userDetail));
+        return response.data;
+        
+    } catch (error) {
+        console.log(error)
+    }
+})
+
+export const signup = createAsyncThunk("/auth/signup", async (userDetail) => {
+    try{
+        const response = await axios.post("/api/auth/signup",JSON.stringify(userDetail))
+        return response.data
+    }
+    catch(error){
+        console.log(error)
+
+    }
+})
+
+
+const authSlice = createSlice({
+    name:"auth",
+    initialState,
+    extraReducers(builder) {
+        builder.addCase(login.fulfilled,(state,action) =>{
+            localStorage.setItem("token",action.payload.encodedToken)
+            state.currentUser = action.payload.foundUser
+            SuccessToast("Login Successful")
```

Comment:
I think actions other than setting state should not be done in reducers,
you can move SuccessToast to thunk function 'try' block like line number 15 before returning data, and ErrorToast to 'catch' block like line number 28 before.

Reply:
Okay, will do that